src/features/applications/mappers/index.ts

```diff
@@ -109,11 +109,257 @@ const getValue = (bytes: string) => {
   }
 }
 
-const argumentPathSeperator = '-'
-const argumentFieldPath = (methodName: string, argumentIndex: number) => `${methodName}${argumentPathSeperator}${argumentIndex}`
-export const extractArgumentIndexFromFieldPath = (path: string) => parseInt(path.split(argumentPathSeperator)[1])
+const argumentPathSeparator = '-'
+const arrayItemPathSeparator = '.' // This must be a . for react hook form to work
+const argumentFieldPath = (methodName: string, argumentIndex: number) => `${methodName}${argumentPathSeparator}${argumentIndex}`
+export const extractArgumentIndexFromFieldPath = (path: string) =>
+  parseInt(path.split(argumentPathSeparator)[1].split(arrayItemPathSeparator)[0])
 
-// TODO: we need to support different sizes of uint
+const getFieldSchema = (type: algosdk.ABIArgumentType, isOptional: boolean): z.ZodTypeAny => {
+  // TODO: confirm isOptional for arrays
+  if (type instanceof algosdk.ABIUintType) {
+    const max = Math.pow(2, type.bitSize) - 1
+    const uintSchema = z.number().min(0).max(max)
+    return numberSchema(isOptional ? uintSchema.optional() : uintSchema)
+  }
+  if (type instanceof algosdk.ABIByteType) {
+    return zfd.numeric(z.number().min(0).max(255))
+  }
+  if (type instanceof algosdk.ABIBoolType) {
+    return isOptional ? z.boolean().optional() : z.boolean()
+  }
+  if (type instanceof algosdk.ABIUfixedType) {
+    // TODO: review the max
+    const max = Math.pow(2, type.bitSize) - 1
+    const precision = type.precision
+    const uintfixedSchema = z
+      .number()
+      .min(0)
+      .max(max)
+      .refine((n) => n === undefined || n.toString().split('.').length === 1 || n.toString().split('.')[1].length <= precision, {
+        message: `Precision must be less than ${precision}`,
+      })
+    return numberSchema(isOptional ? uintfixedSchema.optional() : uintfixedSchema)
+  }
+  if (type instanceof algosdk.ABIArrayStaticType) {
+    if (type.childType instanceof algosdk.ABIByteType) {
+      return isOptional ? zfd.text().optional() : zfd.text()
+    } else {
+      return z.array(getFieldSchema(type.childType, false))
+    }
+  }
+  if (type instanceof algosdk.ABIAddressType) {
+    return isOptional ? zfd.text().optional() : zfd.text()
+  }
+  if (type instanceof algosdk.ABIArrayDynamicType) {
+    if (type.childType instanceof algosdk.ABIByteType) {
+      return isOptional ? zfd.text().optional() : zfd.text()
+    } else {
+      return z.array(
```

Comment:
I had to make a different schema
```
        z.object({
          id: z.string(),
          child: getFieldSchema(type.childType, false),
        })
```
for dynamic array so that it can be supported by `useFieldArray`. For static array and tuple, we don't need to.

Reply:
also, I updated the static array to add length restriction into the static array